Remiworks/Remiworks.Attributes/Initialization/Initializer.cs

```diff
@@ -2,122 +2,143 @@
-                    }
-                    else if (methods.Any(m => m.GetCustomAttributes<CommandAttribute>().Count() > 0))
-                    {
-                        SetUpCommandMethods(type, queueAttribute.QueueName);
-                        _logger.LogInformation($"Initializing commands {type}");
-                    }
+                    continue;
                 }
-            }
-        }
 
-        private bool TypeContainsBothCommandsAndEvents(Type type)
-        {
-            var methods = type.GetMethods();
+                if (TypeContainsBothCommandsAndEvents(type))
+                {
+                    throw new InvalidOperationException(
+                        $"Type '{type}' can't contain both events and commands. Events and commands should not be sent to the same queue.");
+                }
 
-            return
-                methods.Any(m => m.GetCustomAttributes<EventAttribute>().Count() > 0) &&
-                methods.Any(m => m.GetCustomAttributes<CommandAttribute>().Count() > 0);
-        }
+                var methods = type.GetMethods();
 
-        private void SetUpCommandMethods(Type type, string queueName)
-        {
-            Dictionary<string, MethodInfo> commandsWithMethods = GetCommandsWithMethods(type);
-            _busProvider.SetupRpcListeners(queueName, commandsWithMethods.Keys.ToArray(), CreateCommandReceivedCallback(type, commandsWithMethods));
+                if (methods.Any(m => m.GetCustomAttributes<EventAttribute>().Any()))
+                {
+                    SetUpTopicMethods(type, queueAttribute.QueueName);
+                    Logger.LogInformation($"Initializing event {type}");
+                }
+                else if (methods.Any(m => m.GetCustomAttributes<CommandAttribute>().Any()))
+                {
+                    SetUpCommandMethods(type, queueAttribute.QueueName);
+                    Logger.LogInformation($"Initializing commands {type}");
+                }
+            }
         }
 
         private void SetUpTopicMethods(Type type, string queueName)
         {
-            Dictionary<string, MethodInfo> topicsWithMethods = GetTopicsWithMethods(type);
-            _busProvider.CreateTopicsForQueue(queueName, topicsWithMethods.Keys.ToArray());
-            var callback = CreateEventReceivedCallback(type, topicsWithMethods);
-            _busProvider.BasicConsume(queueName, callback);
+            foreach (var topicWithMethod in GetTopicsWithMethods(type))
+            {
+                var parameterType = topicWithMethod.Value
+                    .GetParameters()
+                    .FirstOrDefault()
+                    ?.ParameterType;
+
+                if (parameterType == null)
+                {
+                    var exception = new InvalidOperationException(
+                        $"No parameters could be found for method '{topicWithMethod.Value.Name}' in type '{type}'");
+                    
+                    Logger.LogError(
```

Comment:
Je kunt hier beter een exception throwen, zodat de gebruiker van het framework hier direct iets aan kan doen als het een error is